File: app/src/main/java/com/anokmik/tripassistant/validator/AuthorTextLengthValidator.java
package com.anokmik.tripassistant.validator;

import android.databinding.ObservableBoolean;

public final class AuthorTextLengthValidator extends TextLengthValidator {

    private final ObservableBoolean firstNameValid;
    private final ObservableBoolean lastNameValid;

    private final int minTextLength;

    public AuthorTextLengthValidator(ObservableBoolean firstNameValid, ObservableBoolean lastNameValid) {
        this.firstNameValid = firstNameValid;
        this.lastNameValid = lastNameValid;
        this.minTextLength = 5;
    }

    public boolean validFields(String firstName, String lastName) {
        firstNameValid.set(higherThan(firstName, minTextLength));
        lastNameValid.set(higherThan(lastName, minTextLength));
        return firstNameValid.get() && lastNameValid.get();
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/validator/AuthorTextLengthValidator.kt
package com.anokmik.tripassistant.validator

import android.databinding.ObservableBoolean

class AuthorTextLengthValidator(private val firstNameValid: ObservableBoolean, private val lastNameValid: ObservableBoolean) : TextLengthValidator() {

    private val minTextLength = 5

    fun validFields(firstName: String?, lastName: String?): Boolean {
        firstNameValid.set(higherThan(firstName, minTextLength))
        lastNameValid.set(higherThan(lastName, minTextLength))
        return firstNameValid.get() && lastNameValid.get()
    }

}